openedx/adg/admin_dashboard/performance/tests/test_views.py

```diff
@@ -0,0 +1,147 @@
+"""
+Tests for `performance` app views
+"""
+import pytest
+from django.contrib.auth.models import Group
+from django.core.exceptions import PermissionDenied
+from django.urls import reverse
+
+from openedx.adg.admin_dashboard.performance.views import CourseGradesView
+from openedx.adg.lms.applications.tests.factories import (
+    BusinessLineFactory,
+    MultilingualCourseFactory,
+    MultilingualCourseGroupFactory
+)
+from openedx.core.djangolib.testing.utils import skip_unless_lms
+
+
+@pytest.fixture(name='course_grades_get_request')
+@skip_unless_lms
+def course_grades_view_get_request(request_factory, courses):
+    """
+    Fixture for `CourseGradesView` `get` request.
+    """
+    kwargs = {'course_key_string': courses['test_course1'].id}
+    request = request_factory.get(reverse('course-grades', kwargs=kwargs))
+    return request
+
+
+@pytest.fixture(name='course_grades_post_request')
+@skip_unless_lms
+def course_grades_view_post_request(request_factory, courses):
+    """
+    Fixture for `CourseGradesView` `get` request.
+    """
+    kwargs = {'course_key_string': courses['test_course1'].id}
+    request = request_factory.post(reverse('course-grades', kwargs=kwargs))
+    return request
+
+
+@pytest.mark.django_db
+@skip_unless_lms
+def test_get_course_grades_permission_denied(courses, mocker, course_grades_get_request, user_with_profile):
+    """
+    Test non-authorized user cannot access `CourseGradesView`.
+    """
+    mocker.patch(
+        'openedx.adg.admin_dashboard.performance.views.CourseGradesView._get_course',
+        return_value=courses['test_course1']
+    )
```

Comment:
Why do we need this mock? We're already sending the course id in the kwargs, the `_get_course` will fetch this course anyway, no?

Reply:
Nope, It is raising authentication error for me.